components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2057,6 +2068,34 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
+    protected defaultSpendingLimit = 100;
+    async getSpendingLimitForTeam(ctx: TraceContext, teamId: string): Promise<number | undefined> {
+        const user = this.checkAndBlockUser("getSpendingLimitForTeam");
+        await this.ensureIsUsageBasedFeatureFlagEnabled(user);
+        await this.guardTeamOperation(teamId, "get");
+        const attributionId = AttributionId.render({ kind: "team", teamId });
+        const costCenter = await this.costCenterDB.findById(attributionId);
+        if (costCenter) {
+            return costCenter.spendingLimit;
+        }
+        // On demand creating a cost center for this team
+        await this.costCenterDB.storeEntry({
+            id: attributionId,
+            spendingLimit: this.defaultSpendingLimit,
+        });
```

Comment:
I find the proactive setting of the value a bit strange. I believe this should happen when enabling Usage Based Billing, rather than when getting the spending limit. The problem with this is the API may call getSpendingLimit irrespective of whether you've got Usage based billing enabled and will then create spending limits for teams which don't even have billing enabled.

Reply:
> I find the proactive setting of the value a bit strange.

True. I put it in here, because I recalled it from a discussion. As I understood it, it would solve the default value problem.

Reply:
Should we ignore the unset spending? Should we render `–` instead, if it's not set?

Reply:
Also, @easyCZ, would it help to check if billing is enabled for that team?

Reply:
> I find the proactive setting of the value a bit strange. 

removed